### PR TITLE
Implemented: reports jobs page (#85zrwrcty)

### LINKED ISSUE #439: Add tab for reports jobs
## What is the motivation for adding/enhancing this feature?

Jobs that have enumeration type "REPORT_SYS_JOB" should show up on a dedicated page on the job manager app above the miscellaneous page.

## What are the acceptance criteria?

The page should be labeled: Reports
Icon to use: bar-chart-outline

### PR DESCRIPTION
**Pull request by k2maan**
### Related Issues

Closes #439 

### Screenshots of Visual Changes before/after (If There Are Any)
reports

**IMPORTANT NOTICE** - Remember to add changelog entry

### Contribution and Currently Important Rules Acceptance

- [ ] I read and followed [contribution rules](https://github.com/hotwax/job-manager#contribution-guideline)

### CODE REVIEW

File: src/store/modules/job/actions.ts [hunk 1]
```diff
@@ -887,6 +887,56 @@ const actions: ActionTree<JobState, RootState> = {
...
+          if (job.statusId === 'SERVICE_DRAFT') delete job.runTime;
+          return { ...job, 'status': job.statusId }
+        })
+
+        if (payload.viewIndex > 0) jobs = state.reports.list.concat(jobs);
+
+        const tempExprList = [] as any;
+        const enumIds = [] as any;
+
+        resp.data.docs.map((item: any) => {
```
Comment (ymaheshwari1) [inline src/store/modules/job/actions.ts:925]:
```suggestion
        resp.data.docs.map((job: any) => {
```

File: src/views/Reports.vue [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+            <div class="ion-text-center">
+              <ion-button fill="outline" @click="refreshJobs()">
+                {{ $t('retry') }}
+                <ion-spinner v-if="isRetrying" name="crescent" />
+              </ion-button>
+            </div>
+          </div>
+
+        <div v-else>
+            <ion-list>
```
Comment (ymaheshwari1) [inline src/views/Reports.vue:24]: Fix indentation.

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

File: src/components/Menu.vue [hunk 3]
```diff
@@ -83,7 +83,7 @@ import {
 } from "@ionic/vue";
 import { defineComponent, ref } from "vue";
 import { mapGetters } from "vuex";
-import { pulseOutline, calendarNumberOutline, terminalOutline, ticketOutline, albumsOutline, shirtOutline, settings, iceCreamOutline, libraryOutline } from "ionicons/icons";
+import { albumsOutline, barChartOutline, calendarNumberOutline, iceCreamOutline, libraryOutline, pulseOutline, settings, shirtOutline, terminalOutline, ticketOutline } from "ionicons/icons";
```
Comment (ymaheshwari1) [inline src/components/Menu.vue:86]:
```suggestion
import { albumsOutline, barChartOutline, calendarNumberOutline, iceCreamOutline, libraryOutline, pulseOutline, settingsOutline, shirtOutline, terminalOutline, ticketOutline } from "ionicons/icons";
```

File: src/views/Reports.vue [hunk 4]
```diff
@@ -0,0 +1,170 @@
+<template>
+  <ion-page>
+    <ion-header :translucent="true">
+      <ion-toolbar>
+        <ion-menu-button slot="start" />
+        <ion-title>{{ $t("Reports") }}</ion-title>
```
Comment (ymaheshwari1) [inline src/views/Reports.vue:6]: Make the entry for static text in en.json

File: src/views/Reports.vue [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+      </ion-toolbar>
+    </ion-header>
+
+    <ion-content>
+      <main>
+        <section>
+          <div v-if="reportsJobs?.length === 0">
```
Comment (ymaheshwari1) [inline src/views/Reports.vue:13]:
```suggestion
          <div v-if="!reportsJobs?.length">
```

File: src/store/modules/job/actions.ts [hunk 6]
```diff
@@ -887,6 +887,56 @@ const actions: ActionTree<JobState, RootState> = {
...
+        "shopId_fld0_value": store.state.user.currentShopifyConfig?.shopId,
+        "shopId_fld0_grp": "1",
+        "shopId_fld0_op": "equals",
+        "shopId_fld1_grp": "2",
+        "shopId_fld1_op": "empty"
+      } as any,
+      "fieldList": [ "systemJobEnumId", "runTime", "tempExprId", "parentJobId", "serviceName", "jobId", "jobName", "currentRetryCount", "statusId", "productStoreId", "runtimeDataId", "enumName", "shopId", "description" ],
+      "noConditionFind": "Y",
+      "viewSize": payload.viewSize,
+      "viewIndex": payload.viewIndex,
```
Comment (ymaheshwari1) [inline src/store/modules/job/actions.ts:907]: We should also pass eComStoreId when fetching reports job.

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

File: src/store/modules/job/actions.ts (hunk 6 above)
Comment (k2maan) [reply src/store/modules/job/actions.ts:914]: Improved sir

**Review by adityasharma7 — APPROVED** [review]

**Pull request merged**